app.js

```diff
@@ -1,2 +1,117 @@
-/* add your code here */
+document.addEventListener('DOMContentLoaded', () => {
+    const stocksData = JSON.parse(stockContent);
+    const userData = JSON.parse(userContent);
+
+    const deleteButton = document.querySelector('#btnDelete');
+    const saveButton = document.querySelector('#btnSave');
+
+    generateUserList(userData, stocksData);
+
+    deleteButton.addEventListener('click', (event) => {
+        event.preventDefault();
+
+        const userId = document.querySelector('#userID').value;
+        const userIndex = userData.findIndex(user => user.id == userId);
+        
+        userData.splice(userIndex, 1);
+        generateUserList(userData, stocksData);
+    })
+
+    saveButton.addEventListener('click', (event) => {
+        e.preventDefault();
+        const id = document.querySelector('#userID').value;
+        const userIndex = userData.findIndex(user =>user.id ==id);
+
+        const newUsers = [
+            ...userData.slice(0, userIndex),
+            {
+                ...userData[userIndex],
+                user: {
+                    firstname: document.querySelector('#firstname').value,
+                    lastname: document.querySelector('#lastname').value,
+                    address: document.querySelector('#address').value,
+                    city: document.querySelector('#city').value,
+                    email: document.querySelector('#email').value,
+                },
+            },
+            ...userData.slice(userIndex + 1)
+        ];
+
```

Comment:
Hi @Akulaharipriya  looks like you're missing the variable assignment here. We have to reset the user list with the updated user list:

```js
 userData = newUsers;
 ```